services/src/main/java/io/scalecube/services/metrics/Metrics.java

```diff
@@ -0,0 +1,190 @@
+package io.scalecube.services.metrics;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import com.codahale.metrics.Timer.Context;
+
+import java.lang.reflect.Method;
+
+public class Metrics {
+
+  private final MetricRegistry registry;
+
+  private final MeterFactory meterFactory = new MeterFactory() {
+
+    @Override
+    public Meter get(final String component, final String methodName, final String eventType) {
+      return registry.meter(MetricRegistry.name(component, methodName, eventType));
+    }
+
+    @Override
+    public <T> Meter get(final Class<T> component, final String methodName, final String eventType) {
+      return get(component.getName(), methodName, eventType);
+    }
+  };
+
+  private final TimerFactory timerFactory = new TimerFactory() {
```

Comment:
Frankly, I don't see any value from those separate interfaces for factories. So I propose to remove all this MeterFactory, TimerFactory etc. And instead just have at Metrics class methods like getTimer, getMeter etc.